sources/web/datalab/templates/terminals.html

```diff
@@ -0,0 +1,124 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+  <title>Google Cloud DataLab</title>
+  <link rel="shortcut icon" type="image/x-icon" href="/static/favicon.ico">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+
+  <link rel="stylesheet" href="/static/components/codemirror/lib/codemirror.css" />
+  <link rel="stylesheet" href="/static/style/style.min.css" type="text/css" />
+  <link rel="stylesheet" href="/static/style/datalab.css" type="text/css" />
+  <link rel="stylesheet" id="themeStylesheet" href="/static/style/custom.css" type="text/css" />
+  <link rel="stylesheet" href="/static/terminal/css/override.css" type="text/css">
+
+  <script src="/static/components/jquery/jquery.min.js"></script>
+</head>
+<body class="terminal-app"
+  data-base-url="<%baseUrl%>"
+  data-notebook-path="/"
+  data-ws-path="<%terminalId%>"
+  data-feedback-id="<%feedbackId%>"
+  data-version-id="<%versionId%>"
+  data-signed-in="<%isSignedIn%>"
+  data-user-id="<%userId%>"
+  data-account="<%account%>">
+
+  <div id="app">
+    <div id="appBar">
+    </div>
+    <div id="header"></div>
+    <div id="site">
+      <div id="terminado-container">
+      </div>
+    </div>
+    <div style="position:absolute; left:-1000em">
+      <pre id="dummy-screen" style="border: solid 5px white;" class="terminal">
```

Comment:
It should actually be 25x80, I'll fix that.

The reason these exist (this `pre` and the following `span`) is they're used by the terminal script to measure the correct row height and column width, so that the number of rows and columns can be recalculated on element resize.

Take a look at https://github.com/jupyter/notebook/blob/master/notebook/static/terminal/js/main.js, lines 34 through 57.